core/src/main/java/com/scalar/db/util/ActiveExpiringMap.java

```diff
@@ -16,21 +16,23 @@
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicLong;
-import java.util.function.Consumer;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.ThreadSafe;
 
 @ThreadSafe
 public class ActiveExpiringMap<K, V> {
-  private final ConcurrentMap<K, ValueHolder<V>> map;
+  private final ConcurrentMap<K, ValueHolder> map;
   private final long valueLifetimeMillis;
   private final long valueExpirationThreadIntervalMillis;
-  private final Consumer<V> valueExpirationHandler;
+  private final BiConsumer<K, V> valueExpirationHandler;
```

Comment:
Change the type of `valueExpirationHandler` to `BiConsumer<K, V>` so that the handler can receive the kay of the expired entry in addition to the value.